app/controllers/inclusion_connect_controller.rb

```diff
@@ -21,9 +21,18 @@ def callback
       redirect_to root_path
     else
       flash[:error] = error_message
-      Sentry.capture_message("Failed to authenticate agent with InclusionConnect", fingerprint: ["ic_agent_not_found"])
+      Sentry.capture_message("Failed to authenticate agent with InclusionConnect", fingerprint: ["ic_other_error"])
       redirect_to new_agent_session_path
     end
+  rescue InclusionConnect::AgentNotFoundError => e
+    flash[:error] = "Aucun agent #{current_domain.name} n'existe pour l'email Inclusion Connect suivant : #{e.message}. Vous devez être invité par un agent existant pour vous connecter. " \
```

Comment:
On pourrait alléger le texte pour gagner en lisibilité :
```suggestion
    flash[:error] = "Il n'y a pas de compte pour l'adresse mail {e.message}. Vous devez être invité par un agent existant pour vous connecter. " \
```

Reply:
Par ailleurs, là on met un texte d'aide pour les gens qui essayent de se connecter alors qu'ils n'ont pas été invité, mais est-ce qu'il ne faudrait pas plutôt qu'on gère le cas des gens qui essayent de se connecter avec la mauvaise adresse ? Par exemple en remplaçant la deuxième phrase avec un texte du genre `"Vous devez utiliser Inclusion Connect avec l'adresse mail à laquelle vous avez reçu votre invitation sur #{current_domain.name}"`
En regardant la liste des emails pour lesquels on a eu cette erreur, je pense qu'on peut évaluer le scénario le plus fréquent (je suis chaud pour aider à investiguer ça)

Reply:
J'ai regardé les emails et on a plusieurs scénarios :
- Le cas ou je ne retrouve aucune info sur l'agent en utilisant son nom ou une partie de l'email qu'il a dans IC.
- Le cas ou on peut retrouver des infos en utilisant une partie de l'email (ex il a utilisé un email d'un domaine différent : `@departementXX.fr` alors qu'il a été invité sur `@conseiller-numerique.fr`)
- Le cas ou il y a une faute dans le nom de l'agent qui a été invité ex : `caoline.qqchose@` au lieu de `caroline.qqchose@` dans inclusion connect.

Reply:
Je pense que c'est effectivement plus parlant avec ta proposition. Je fais les changements.

Reply:
merci d'avoir regardé les emails, c'est super intéressant et ça fait du bien de savoir qu'on propose une interface  qui répond vraiment au problème 😃